script/controller/tasklist-front.js

```diff
@@ -4,42 +4,66 @@
 
 angular.module('tasklist-front', ['tasklist-back', 'users-back', 'utils'])
 
-    .controller('widgetController', function ($scope, $state, Tasks, Users, Utils) {
+    .controller('widgetController', function ($scope, $state, TaskItem, Tasks, Users, Utils, GoogleCalendarService) {
 
         $scope.description = "";
         $scope.assignee = "";
-        $scope.assignedByMe = $state.is('tasks.assignedByMe');
+        $scope.properties = {
+            list: $state.current.name
+        };
 
         $scope.items = Tasks.getItems();
 
-        $scope.items.forEach(function (item) {
-            item.hovered = false;
-            item.focused = false;
-        });
+        /* TODO refactor */
+        if (!Users.isLocalUser()) {
+            GoogleCalendarService.fetchTasks().then(function (tasks) {
+                tasks.forEach(function (task) {
+                    var ti = new TaskItem();
+                    ti.restoreFrom(task);
+
+                    if (!Users.isExistent(ti.getAuthor())) {
+                        Users.add(ti.getAuthor());
+                    }
 
-        function updateItem(item) {
+                    $scope.addTask(ti);
+                });
+                $scope.items = Tasks.getItems();
+            });
+        }
+        /**/
+
+        $scope.updateItem = function (item) {
             var index = $scope.items.lastIndexOf(item);
             $scope.items[index] = Tasks.getItem(item.task.id);
-        }
+            console.log($scope.items);
+        };
 
-        function updateItemAssignee(item) {
+        $scope.updateItemAssignee = function (item) {
             var index = $scope.items.lastIndexOf(item);
             var validItem = Tasks.getItem(item.task.id);
             $scope.items[index].task.assignee = validItem.task.assignee;
             $scope.items[index].access = validItem.access;
-        }
+        };
 
-        function updateItemDescription(item) {
+        $scope.updateItemDescription = function (item) {
             var index = $scope.items.lastIndexOf(item);
             $scope.items[index].task.description = Tasks.getItem(item.task.id).task.description;
-        }
+        };
 
-        $scope.addTask = function () {
-            Tasks.addTask($scope.description, Users.getCurrentUser(), $scope.assignee);
+        $scope.uiAddTask = function () {
```

Comment:
bad naming (ui prefix)
it could be gotoAddTask, initializeAddTaskPanel, initTask etc